File: src/main/scala/org/xbery/artbeams/users/admin/UserForm.scala
package org.xbery.artbeams.users.admin

import net.formio.{Field, FormMapping, Forms}
import org.xbery.artbeams.common.form.ScalaForms
import org.xbery.artbeams.users.domain.EditedUser

/**
  * User edit form.
  * @author Radek Beran
  */
class UserForm {
  lazy val definition: FormMapping[EditedUser] = {
    Forms.basic(classOf[EditedUser], "user")
      .field("id", Field.HIDDEN)
      .field("login", Field.TEXT)
      .field("password", Field.PASSWORD)
      .field("password2", Field.PASSWORD)
      .field("firstName", Field.TEXT)
      .field("lastName", Field.TEXT)
      .field("email", Field.TEXT)
      .field("roleIds")
      .build(ScalaForms.CzConfig)
  }
}

File: src/main/kotlin/org/xbery/artbeams/users/admin/UserForm.kt
package org.xbery.artbeams.users.admin

import net.formio.Field
import net.formio.FormMapping
import net.formio.Forms
import org.xbery.artbeams.common.form.FormUtils
import org.xbery.artbeams.users.domain.EditedUser

/**
 * User edit form.
 * @author Radek Beran
 */
open class UserForm {
    companion object {
        val definition: FormMapping<EditedUser> =
            Forms.basic(EditedUser::class.java, "user")
                .field<String>("id", Field.HIDDEN)
                .field<String>("login", Field.TEXT)
                .field<String>("password", Field.PASSWORD)
                .field<String>("password2", Field.PASSWORD)
                .field<String>("firstName", Field.TEXT)
                .field<String>("lastName", Field.TEXT)
                .field<String>("email", Field.TEXT)
                .field<List<String>>("roleIds")
                .build(FormUtils.CzConfig)
    }
}
